mixnet/util/src/lib.rs

```diff
@@ -0,0 +1,26 @@
+use std::{collections::HashMap, net::SocketAddr, sync::Arc};
+
+use parking_lot::Mutex;
+use tokio::net::TcpStream;
+
+#[derive(Clone)]
+#[repr(transparent)]
+pub struct ConnectionCache {
+    cache: Arc<Mutex<HashMap<SocketAddr, Arc<tokio::sync::Mutex<TcpStream>>>>>,
+}
```

Comment:
Now I think we can rename this because this is not a cache anymore :)

As we discussed, I'm fine with both hashmap and LRU at this moment, because they have pros and cons and we don't have any exact plan how to compose the mixnet topology yet :)

Reply:
In my opinion, one thing we can easily do at next PRs is adding the `max_connections` config to mixnode, so that we can cover the case when there are so many mixclients trying connect to the mixnode. Let's discuss soon.